### PR TITLE
test: Mock the dom

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
This technically doesn't require changes to our production code, but specifying the CSS selectors to mock might be much easier and stable when there are dedicated classes or ids in the DOM.

File: test/common/chromium-cdp-driver.js [hunk 1]
```diff
@@ -204,6 +204,17 @@ function setupLocalFunctions(client) {
             client.Page.reload(args);
         });
     };
+
+    async function setCSS({ text, frame }) {
+        console.error("SET CSS", text, frame, frameNameToFrameId[frame]);
```
Comment (mvollmer) [inline test/common/chromium-cdp-driver.js:209]: Debug leftover.

File: test/common/testlib.py [hunk 2]
```diff
@@ -332,6 +332,37 @@ def call_js_func(self, func: str, *args: Any) -> Optional[Any]:
         """
         return self.eval_js("%s(%s)" % (func, ','.join(map(jsquote, args))))
 
+    def set_css(self, text: str):
+        """Set extra CSS rules for the current frame
+
+        Each call to this function replaces the CSS that h
```
Comment (mvollmer) [inline test/common/testlib.py:338]: Rest of comment got lost somehow.

**Comment by mvollmer** [comment]
And kudos to @garrett for figuring out the CSS tricks!

**Comment by martinpitt** [comment]
Nice! However, I'm a bit cautious with the no-test. I triggered the /firefox and /pybridge scenarios which also run pixel tests. They did not actually appear here and probably won't start (GitHub is currently kaputt), if not I'll re-trigger them tomorrow morning.

**Review by martinpitt — APPROVED** [review]
"Mock the Dom" really sounds like [Hack the Planet](https://www.youtube.com/watch?v=EPHPu4PV-Bw)! :notes: :drum: 🤘 !

Seems to work fine, great work you two!

**Pull request merged**